### PR TITLE
[main] Verify promotion tag for patch releases

### PR DESCRIPTION
**Pull request by pierDipi**
Backport of https://github.com/openshift-knative/serverless-operator/pull/2755

### CODE REVIEW

File: .github/workflows/validate-release-promotion.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              count=$(( count + 1 ))
+              echo "File: $file"
+              echo "promotion.to[0].name: $promotion_tag"
+              echo
+            fi
+          done <   <(find ci-operator/config/openshift-knative/serverless-operator -type f -name "*${release}*" -print0)
+          
+          if [ $count -ne 0 ]; then
+            echo "## $count files with an unexpected promotion name or tag"
+            echo "1. Use the [release-generate-ci](https://github.com/openshift-knative/serverless-operator/actions/workflows/release-generate-ci.yaml?query=branch%3Arelease-1.33) GH actions workflow in serverless-operator on the $release branch with inputs:"
```
Comment (mgencur) [inline .github/workflows/validate-release-promotion.yaml:82]: Should this query point to release-1.33? This PR is against main

Comment (pierDipi) [reply .github/workflows/validate-release-promotion.yaml]: Added it as a variable

**Pull request merged**